src/lib/component/toast/ExecuteCheckStatusToast.js

```diff
@@ -1,102 +1,114 @@
-    this._willUnmount = true;
-  }
+import { InlineLink, Toast } from "/lib/component/base";
 
-  render() {
-    const { mutation, onClose, checkName, entityName, namespace } = this.props;
-    const { loading } = this.state;
+const ExecuteCheckStatusToast = ({
+  onClose,
+  resolved,
+  rejected,
+  checkName,
+  entityName,
+  namespace,
+}) => {
+  const subject = (
+    <React.Fragment>
+      <strong>{checkName}</strong>
+      {entityName && (
+        <span>
+          {" "}
+          on <strong>{entityName}</strong>
+        </span>
+      )}
+    </React.Fragment>
+  );
 
-    const subject = (
-      <React.Fragment>
-        <strong>{checkName}</strong>
-        {entityName && (
+  if (resolved) {
+    return (
+      <Toast
+        maxAge={5000}
+        variant="success"
+        message={
           <span>
-            {" "}
-            on <strong>{entityName}</strong>
+            Done executing {subject}.{" "}
+            <InlineLink
+              component={NamespaceLink}
+              namespace={namespace}
+              to={`/events?filter=${encodeURIComponent(
+                `check.name === "${checkName}"${
+                  entityName ? ` && entity.name === "${entityName}"` : ""
+                }`,
+              )}`}
+            >
+              View&nbsp;events.
+            </InlineLink>
           </span>
-        )}
-      </React.Fragment>
+        }
+        onClose={onClose}
+      />
     );
+  }
 
+  if (rejected) {
     return (
-      <ToastConnector>
-        {({ setToast }) => (
-          <Toast
-            maxAge={loading ? undefined : 5000}
-            variant={loading ? "info" : "success"}
-            progress={loading && <LinearProgress />}
-            message={
-              loading ? (
-                <span>Executing {subject}.</span>
-              ) : (
-                <span>
-                  Done executing {subject}.{" "}
-                  <InlineLink
-                    component={NamespaceLink}
-                    namespace={namespace}
-                    to={`/events?filter=${encodeURIComponent(
-                      `check.name === "${checkName}"${
-                        entityName ? ` && entity.name === "${entityName}"` : ""
-                      }`,
-                    )}`}
-                  >
-                    View&nbsp;events.
-                  </InlineLink>
-                </span>
-              )
-            }
-            onClose={() => {
-              onClose();
-
-              if (loading) {
-                const onMutationEnd = () =>
-                  setToast(undefined, ({ remove }) => (
-                    <ExecuteCheckStatusToast {...this.props} onClose={remove} />
-                  ));
-                mutation.then(onMutationEnd, onMutationEnd);
-              }
-            }}
-          />
-        )}
-      </ToastConnector>
+      <Toast
+        variant="error"
+        // TODO: Render error details.
```

Comment:
We'll be doing this in a later PR?

Reply:
Yeah, this is deferred until we have some further product-level discussion.